### DISCUSSION TITLE
JWT token lifetime?

### DISCUSSION THREAD

**Discussion by CtFelix** [strapi/strapi#7043] [Legacy Topics]
Hi everyone!

I can't find any information about JWT token lifetime, how long in will be valid?

And how i can automatically refresh the token when it will be not valid?

And how can i do token validation, before i get exeption about token lifetime end? What is name of endpoit link?

**Comment by MattieBelt** [comment]
Hi @CtFelix, you can configure the token [here](https://strapi.io/documentation/v3.x/plugins/users-permissions.html#jwt-configuration).

Token refreshing isn't supported right now #1676.

You could customize the token validation, with an [plugin extension](https://strapi.io/documentation/v3.x/concepts/customization.html#plugin-extensions).

**Reply by CtFelix** [reply]
Hi @MattieBelt, thank you!
it's that i needed

**Reply by CtFelix** [reply]
> Hi @CtFelix, you can configure the token [here](https://strapi.io/documentation/v3.x/plugins/users-permissions.html#jwt-configuration).

it's i can use for some time, it's really help me

> Token refreshing isn't supported right now #1676.

is's so sad :)

> You could customize the token validation, with an [plugin extension](https://strapi.io/documentation/v3.x/concepts/customization.html#plugin-extensions).

may be, one day, when i start learning and coding on JS i will do it, but it's not sure ))
i hope, someone make it earlier :)
